README.md

```diff
@@ -4,16 +4,16 @@ This repository contains infrastructure necessary to process cross-chain executi
 The codebase uses [Foundry](https://github.com/foundry-rs/foundry) as the development framework. In order to run tests, deploy contracts, or perform other operations, use standard [Foundry](https://github.com/foundry-rs/foundry) commands.
 
 ## ⚙️ Components
-### 🔊 Forwarders
-Forwarders are contracts on the host domain, where the governance of the protocol resides. <br>These contracts abstract away all complexity of using a bridge and facilitate seamless message passing between domains. <br>Host domain admin uses forwarders to trigger payload executions on a bridged domain instances of the protocol.
 ### 🚦 Executors
 Executors serve as admins of the bridged domain instances of the protocol. <br>They are responsible for storying the queue of proposals passed from the host domain governance and their execution. <br> They manage bridged domain protocol instance using standard payload pattern.
 ## ✍️ Architecture Diagram
 ![Architecture Diagram](/diagram.png)
 ## 🤝 Contribution Guidelines
 In order to add governance relay infrastructure for a new domain, perform the following steps:
-1. Go to [XChain Helpers](https://github.com/marsfoundation/xchain-helpers) repository and add a proper domain helper abstracting away the process of passing messages between host domain and your bridged domain.
-2. Add proper `CrosschainForwarder` to the `/src/forwarders` directory and proper `BridgeExecutor` to the `/src/executors` directory. Follow currently used naming convention.
+1. Go to [XChain Helpers](https://github.com/marsfoundation/xchain-helpers) repository and add:
+    1. A domain helper abstracting away the process of passing messages between host domain and your bridged domain.
+    2. A helper function to the XChainForwarders library abstracting away host domain interactions with the bridge.
+2. Add `BridgeExecutor` to the `/src/executors` directory. Follow currently used naming convention.
 3. If there are any dependencies required, copy them into `/src/dependencies/<YOUR_DOMAIN_NAME>` directory.
```

Comment:
I would just drop this directory and inline all remaining interfaces/libraries. `AddressAliasHelper` function that is used can just be added to the Executor as an internal function imo.

Reply:
Done ✅